incubator/hnc/test/conversion/conversion_test.go

```diff
@@ -0,0 +1,144 @@
+		MustRun("kubectl hns set", nsC, "--parent", nsA)
+
+		// Verify kubectl-hns plugin works and 'b' and 'c' are children of 'a'.
+		RunShouldContainMultiple([]string{nsB, nsC}, 2, "kubectl hns tree", nsA)
+		// Verify there's no 'v1alpha2' in all three CRDs for now.
+		checkCRDVersionInField("v1alpha2", ".spec", crds, false,2)
+		checkCRDVersionInField("v1alpha2", ".status", crds, false,2)
+
+		// Deploy the HEAD to include API conversion and install v1alpha2 kubectl plugin.
+		GinkgoT().Log("Set up v1alpha2")
+		setupV1alpha2()
+		// Wait for the cert rotator to write caBundles in CRD conversion webhooks.
+		ensureCRDConvWHReady(crds)
+
+		// Verify the conversion works with 'v1alpha1' still in the spec.versions but not in status.storedVersions.
+		checkCRDVersionInField("v1alpha1", ".spec.versions", crds, true, 2)
+		checkCRDVersionInField("v1alpha2", ".spec.versions", crds, true, 2)
+		// Give this a longer timeout for the reconciler to remove v1alpha1 from storedVersions.
+		checkCRDVersionInField("v1alpha1", ".status.storedVersions", crds, false, 5)
+		checkCRDVersionInField("v1alpha2", ".status.storedVersions", crds, true,2)
+
+		// Verify v1alpha2 kubectl-hns plugin still works
+		RunShouldContainMultiple([]string{nsB, nsC}, 2, "kubectl hns tree", nsA)
+		// Verify cascading deletion still works.
+		MustRun("kubectl hns set", nsA, "-a")
+		MustRun("kubectl delete ns", nsA)
+		// Verify conditions still work.
+		RunShouldContain("CritParentMissing", 2, "kubectl hns describe", nsC)
+	})
+})
+
+// Install HNC and kubectl plugin with v1alpha1.
+func setupV1alpha1(hncVersion string){
+	goPath := os.Getenv("GOPATH")
+	MustRun("kubectl apply -f https://github.com/kubernetes-sigs/multi-tenancy/releases/download/hnc-"+hncVersion+"/hnc-manager.yaml")
+	MustRun("curl -L https://github.com/kubernetes-sigs/multi-tenancy/releases/download/hnc-"+hncVersion+"/kubectl-hns -o "+goPath+"/bin/kubectl-hns")
+	MustRun("chmod +x "+goPath+"/bin/kubectl-hns")
+}
+
+// Install HNC and kubectl plugin with v1alpha2.
+func setupV1alpha2(){
+	goPath := os.Getenv("GOPATH")
+	MustRun("kubectl apply -f ../../manifests/hnc-manager.yaml")
+	MustRun("rm -r "+goPath+"/bin/kubectl-hns")
+	MustRun("cp ../../bin/kubectl/kubectl-hns "+goPath+"/bin/kubectl-hns")
+}
+
+// Check if a specific version exists in the field as expected for a list of CRDs.
+func checkCRDVersionInField(version, field string, crds []string, expected bool, seconds float64) {
+	for _, crd := range crds {
+		if expected {
+			RunShouldContain(version, seconds, "kubectl get -o template crd/"+crd+" --template={{"+field+"}}")
+		} else {
+			RunShouldNotContain(version, seconds, "kubectl get -o template crd/"+crd+" --template={{"+field+"}}")
+		}
+	}
+}
+
+// ensureHNCCleared ensures the namespace and the CRDs are gone.
+func ensureHNCCleared(crds []string){
+	RunShouldNotContain("hnc-system", 10, "kubectl get ns")
+	for _, crd := range crds {
+		RunShouldNotContain(crd, 10, "kubectl get crd")
+	}
```

Comment:
All the CRDs have the name "hnc" in them, so you could just say: `RunShouldNotContain("hnc", 10, "kubectl get crds")`

Reply:
Thanks for the suggestion this simplifies it a lot!